### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by heartwilltell**
Remove unneeded newlines. Fix type for consts BackupStarting, BackupRunning, BackupFailed, BackupSucceeded from string to PXCBackupState. The same for list of const with type BcpRestoreStates. Changed fmt.Errorf("%v", err) to fmt.Errorf("%w", err) to wrap errors. Used more idiomatic errors.Is instead of switch err == AnotherErr. Formatted imports with Goimport.

Signed-off-by: Serhii Mariiekha <heartwilltell@gmail.com>

### CODE REVIEW

File: pkg/controller/pxcbackup/controller.go [hunk 1]
```diff
@@ -224,7 +224,7 @@ func (r *ReconcilePerconaXtraDBClusterBackup) getClusterConfig(cr *api.PerconaXt
 		availableClusters = append(availableClusters, cluster.Name)
 	}
 
-	return nil, fmt.Errorf("wrong cluster name: %q. Clusters avaliable: %q", cr.Spec.PXCCluster, availableClusters)
+	return nil, fmt.Errorf("wrong cluster name: %q. Clusters `available: %q", cr.Spec.PXCCluster, availableClusters)
```
Comment (delgod) [inline pkg/controller/pxcbackup/controller.go:227]:
```suggestion
	return nil, fmt.Errorf("wrong cluster name: %q. Clusters available: %q", cr.Spec.PXCCluster, availableClusters)
```

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/pxcbackup/controller.go (hunk 1 above)
Comment (heartwilltell) [reply pkg/controller/pxcbackup/controller.go]: Fixed 👍

File: pkg/apis/pxc/v1/pxc_backup_types.go [hunk 2]
```diff
@@ -43,10 +43,10 @@ type PXCBackupState string
...
 	BackupNew       PXCBackupState = ""
-	BackupStarting                 = "Starting"
-	BackupRunning                  = "Running"
-	BackupFailed                   = "Failed"
-	BackupSucceeded                = "Succeeded"
+	BackupStarting  PXCBackupState = "Starting"
+	BackupRunning   PXCBackupState = "Running"
+	BackupFailed    PXCBackupState = "Failed"
+	BackupSucceeded PXCBackupState = "Succeeded"
 )
```
Comment (heartwilltell) [inline pkg/apis/pxc/v1/pxc_backup_types.go:51]: Previously only BackupNew had a type of PXCBackupState. Other const had string type.

Comment (heartwilltell) [reply pkg/apis/pxc/v1/pxc_backup_types.go:51]: Now all consts from this block has PXCBackupState type

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| upgrade-haproxy | passed |
| scaling | passed |
| recreate | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| security-context | passed |
| demand-backup | passed |
| pitr | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | failed |
| scheduled-backup | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/788/commits/714170f0108866467e3998c897c11c25c64dccdb
image: `perconalab/percona-xtradb-cluster-operator:PR-788-714170f0`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/788/commits/714170f0108866467e3998c897c11c25c64dccdb
image: `perconalab/percona-xtradb-cluster-operator:PR-788-714170f0`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| storage | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| scaling | passed |
| demand-backup | passed |
| scaling-proxysql | passed |
| pitr | passed |
| security-context | passed |
| demand-backup-encrypted-with-tls | passed |
| upgrade-proxysql | passed |
| self-healing | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| init-deploy | passed |
| self-healing-advanced | passed |
| limits | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/788/commits/714170f0108866467e3998c897c11c25c64dccdb
image: `perconalab/percona-xtradb-cluster-operator:PR-788-714170f0`

File: go.sum [hunk 3]
```diff
@@ -1,4 +1,6 @@
+bazil.org/fuse v0.0.0-20160811212531-371fbbdaa898 h1:SC+c6A1qTFstO9qmB86mPV2IpYme/2ZoEQ0hrP+wo+Q=
```
Comment (qJkee) [inline go.sum:1]: Why this file is changed that much?

File: vendor/modules.txt [hunk 4]
```diff
@@ -10,6 +10,8 @@ github.com/PuerkitoBio/urlesc
 github.com/asaskevich/govalidator
 # github.com/beorn7/perks v1.0.1
 github.com/beorn7/perks/quantile
+# github.com/caarlos0/env v3.5.0+incompatible
```
Comment (qJkee) [inline vendor/modules.txt:13]: it's not added anywhere

**Review by qJkee — CHANGES_REQUESTED** [review]

File: vendor/modules.txt (hunk 4 above)
Comment (heartwilltell) [reply vendor/modules.txt:13]: This dependency is declared here https://github.com/percona/percona-xtradb-cluster-operator/blob/main/cmd/pitr/main.go#L12 and it wasn't in go.mod file, so I added it.
The go.sum file changed automatically when you add or remove something from go.mod file.

File: go.sum (hunk 3 above)
Comment (heartwilltell) [reply go.sum:1]: The go.sum file changed automatically when you add or remove something from go.mod file. Also when you run `go mod tidy` to clean up the mess from modules, the go.sum file also could be changed.

**Review by qJkee — APPROVED** [review]

**Pull request merged**